src/plugins/geoip.server.js

```diff
@@ -1,14 +1,38 @@
+/**
+ * Parse the accepted language header and check if the first languages are from HK
+ * @param {string|undefined} acceptedLanguage Browser accepted language header
+ * @returns {string|undefined}
+ */
+function detectHKBrowserLang(acceptedLanguage) {
+  if (acceptedLanguage) {
+    const langs = acceptedLanguage.split(',');
+    const len = langs.length > 3 ? 3 : langs.length; // only three first browser language are important
+    for (let i = 0; i < len; i += 1) {
+      const pair = langs[i].split(';');
+      if (pair[0].endsWith('-HK')) {
+        // This check if the language is for HK region
+        return 'HK';
+      }
+    }
+  }
+  return undefined;
+}
+
 export default ({ app, store, req, res, query }) => {
   // gcloud magic geoip headers
   // https://cloud.google.com/appengine/docs/standard/go/reference/request-response-headers#app_engine-specific_headers
   const ipCountry = req.headers['x-appengine-country'];
   const ipCity = req.headers['x-appengine-city'];
   const cacheServerName = req.headers['x-forwarded-server'] || '';
-  if ((ipCountry && ipCountry !== 'ZZ') || ipCity || cacheServerName) {
-    const isHK =
-      ipCountry === 'HK' ||
-      ipCity === 'hong kong' ||
-      cacheServerName.includes('HKG');
-    store.dispatch('setIsHK', isHK);
+  const browserLanguage = detectHKBrowserLang(req.headers['accept-language']); // browser language workaround
+  // Only the cache server of the HKG region causes problem to detect people from HK
+  if (!cacheServerName.includes('HKG')) {
```

Comment:
Maybe we can just totally ignore the `cacheServerName`  in this whole logic, and just detect if the country is `ZZ` (which means the geoip has invalid value) ?

Reply:
The thing is the ZZ is not an invalid value, it's just saying that the geoip server can't find the country this ip belong to. In the end even if you check it again with the endpoint provided it will give you back a ZZ again.
In this use case we just want to detect HK and the only reason when we aren't sure if it's an HK user because the geoip might be taken from a cache is when the HKG is present in the cacheServer which is why i think it's better to keep it there. The HKG is cache for user in the hong kong area only from what i understood, so with that in mind the only one that needs to get a call to make sure they aren't from HK is those having it, the others are sure not to be HK user so we'll avoid all the other webcall from cached user outside the HK cache area

Reply:
But the problem here is that we could be accessing the website either with or without firebase cdn, and the latter one almost always return `ZZ`. If the `ZZ` returned is due to using CDN, the next step we would do is calling the geoip API above without CDN, which should yield us something other than `ZZ` (hopefully). 

So ignoring cache region, and just check for `ZZ` as the "check again" flag seems more simple in case we want to check for region other than HK later.

Reply:
For checking region other than HK yes. I was just talking about what is being done now but it's fine for me to change it
So tonight i'll check if it's (ipCountry && ipCountry !== 'ZZ') then i check the variables otherwise i will make the call on the frontend

Reply:
Yes these 2 should be all the change needed, thanks a lot for the work